### PR TITLE
Added toTransform conversion function

### PR DESCRIPTION
**Pull request by DivyanshiVashist**
Converts independent transform properties into a valid string for transform.

### CODE REVIEW

File: toTransform.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<!DOCTYPE html>
+
+<head>
+	<script src="src/toTransform.js"></script>
+</head>
```
Comment (alancutter) [inline toTransform.html]: No need for this file.

File: test/toTransform.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  test('testTranslate', function () {
+    assert.throws(function () {
+      toTransform({translate: ''});
+    }, InvalidArgument);
+
+    assert.equal(toTransform({translate: 'none'}), '');
+
+    assert.throws(function () {
+      toTransform({translate: '20px 30px 27px 90px'});
+    }, InvalidArgument);
```
Comment (alancutter) [inline test/toTransform.js]: We should group these assert.throws checks together to make this easier to read.

File: test/toTransform.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    assert.equal(toTransform({translate: '20px 30px'}), 'translate(20px, 30px)');
```
Comment (alancutter) [inline test/toTransform.js]: Add tests for different numbers of inputs.

File: test/toTransform.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  });
```
Comment (alancutter) [inline test/toTransform.js]: Add a test for random garbage input.

File: src/toTransform.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      for (var counter = 0; counter < numValues - 1; counter++) {
+        result += valuesArray[counter] + ' ';
+      }
+
+      result += valuesArray[counter] + ')';
+    }
+    return result;
+  }
+
+  function checkScale (properties) {
```
Comment (alancutter) [inline src/toTransform.js]: I would name these convertX() etc instead of checkX().

File: src/toTransform.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+    return result;
+  }
+
+  window.toTransform = function (properties) {
+    var result = '';
+
+    result += checkTranslate(properties);
+    result += checkRotate(properties);
+    result += checkScale(properties);
```
Comment (alancutter) [inline src/toTransform.js]: We should not check in untested code. Let's leave rotate and scale for separate PRs or state that this is a WIP if you intend to add tests to this PR later.

File: src/toTransform.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
+(function () {
+  function InvalidArgument (message) {
+    this.message = message;
+    this.name = 'InvalidArgument';
+  }
+
+  function checkTranslate (properties) {
+    var result = '';
+    if (properties.translate !== undefined && properties.translate !== 'none') {
```
Comment (alancutter) [inline src/toTransform.js]: Prefer early returns over long if bodies.
E.g.
```
if (...)
  return '';
```

File: src/toTransform.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+      if (numValues === 1 && valuesArray[0] === '') {
+        throw new InvalidArgument('Empty string is not a valid argument');
+      }
+
+      for (var counter = 0; counter < numValues - 1; counter++) {
+        result += valuesArray[counter] + ', ';
+      }
+
+      result += valuesArray[counter] + ')';
```
Comment (alancutter) [inline src/toTransform.js]: Shorter version: `result += valuesArray.join(', ') + ')'`

**Comment by alancutter** [comment]
Please make the title/description more descriptive about what this PR is changing.

File: toTransform.html (hunk 1 above)
Comment (DivyanshiVashist) [inline toTransform.html]: Done :tada:

File: test/toTransform.js (hunk 2 above)
Comment (DivyanshiVashist) [inline test/toTransform.js]: Done :tada:

File: test/toTransform.js (hunk 3 above)
Comment (DivyanshiVashist) [inline test/toTransform.js]: Done :rabbit:

File: test/toTransform.js (hunk 4 above)
Comment (DivyanshiVashist) [inline test/toTransform.js]: Done :v:

File: src/toTransform.js (hunk 5 above)
Comment (DivyanshiVashist) [inline src/toTransform.js]: Done :balloon:

File: src/toTransform.js (hunk 6 above)
Comment (DivyanshiVashist) [inline src/toTransform.js]: Done :panda_face:

File: toTransform.html [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<!DOCTYPE html>
+
+<head></head>
```
Comment (alancutter) [inline toTransform.html]: This file is not necessary.

File: src/toTransform.js [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+    var result = '';
+
+    result += convertTranslate(properties);
+    result += convertRotate(properties); // WIP
+    result += convertScale(properties); // WIP
+
+    return result;
+  };
+
+  window.InvalidArgument = InvalidArgument;
```
Comment (alancutter) [inline src/toTransform.js]: We should have `window.toTransform = toTransform;` to make exported methods visually clear.

**Comment by alancutter** [comment]
The description is missing the most important part "Added toTransform conversion function".

File: src/toTransform.js [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+  /**
+  * This function is a WIP.
+  */
+  function convertRotate (properties) {
+    var result = '';
+    if (!(properties.rotate !== undefined && properties.rotate !== 'none')) {
+      return result;
+    }
+
+    if (properties.translate !== undefined && properties.translate !== 'none') {
```
Comment (ewilligers) [inline src/toTransform.js]: rotate.

File: src/toTransform.js [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    result += valuesArray.join(', ') + ')';
+
+    return result;
+  }
+
+  /**
+  * This function is a WIP.
+  */
```
Comment (alancutter) [inline src/toTransform.js]: Let's not check in code that's not ready, remove it from this patch, you can always retrieve it later from your branch's commits.

File: src/toTransform.js [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+  function InvalidArgument (message) {
+    this.message = message;
+    this.name = 'InvalidArgument';
+  }
+
+  function isNumeric (number) {
+    return !isNaN(number);
+  }
+
+  function convertTranslate (properties) {
```
Comment (alancutter) [inline src/toTransform.js]: These conversion functions should just take their respective property's value. They should operate independently from the other properties.

File: src/toTransform.js [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+      // Check if characters before 'px' are valid numbers
+      var number = valuesArray[i].substring(0, numValues - 2);
+      if (!isNumeric(number)) {
+        throw new InvalidArgument('Argument must be a number');
+      }
+    }
+
+    result += valuesArray.join(', ') + ')';
+
+    return result;
```
Comment (alancutter) [inline src/toTransform.js]: I don't think we need to build a result string, just return the equivalent string concatenation expression.

File: src/toTransform.js [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+      // Check if units are px
+      var unit = valuesArray[i].substring(elementLength - 2, elementLength);
+      if (unit !== 'px') {
+        throw new InvalidArgument('Incorrect units for translate');
+      }
+
+      // Check if characters before 'px' are valid numbers
+      var number = valuesArray[i].substring(0, numValues - 2);
+      if (!isNumeric(number)) {
+        throw new InvalidArgument('Argument must be a number');
```
Comment (alancutter) [inline src/toTransform.js]: Are these validation errors tested? Check whether removing them makes tests fail.

File: src/toTransform.js [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+      }
+    }
+
+    return 'translate(' + valuesArray.join(', ') + ')';
+  }
+
+  function toTransform (properties) {
+    var result = '';
+
+    if (properties.translate !== undefined && properties.translate !== 'none') {
```
Comment (alancutter) [inline src/toTransform.js]: I think it makes more sense for this check to be inside convertTranslate, it would just return ''.

**Pull request merged**